tests/middleware/test_message_state.py

```diff
@@ -127,3 +129,12 @@ def do_work(x):
         args = state_backend.get_state(msg.message_id).args
         # if the max_size == 0, then should not storage nothing
         assert list(args) == expected
+
+    def test_set_group_id_in_middleware(self, stub_broker, state_middleware, do_work):
```

Comment:
You can do something simpler, a bit like `test_pending_state_message`  but with a group.